src/cargo/ops/cargo_new.rs

```diff
@@ -85,6 +97,76 @@ struct CargoNewConfig {
     version_control: Option<VersionControl>,
 }
 
+enum TemplateDirectory{
+    Temp(TempDir),
+    Normal(PathBuf)
+}
+
+impl TemplateDirectory {
+    fn path(&self) -> &Path {
+        match *self {
+            TemplateDirectory::Temp(ref tempdir) => tempdir.path(),
+            TemplateDirectory::Normal(ref path) => path.as_path()
+        }
+    }
+}
+
+fn get_input_template(config: &Config, opts: &MkOptions) -> CargoResult<TemplateDirectory> {
+    let path = opts.path;
+    let name = opts.name;
+    let templates_dir = config.template_path();
+    if fs::metadata(&templates_dir).is_err() {
+        try!(fs::create_dir_all(&templates_dir));
+    }
+
+    let template_dir = match (opts.template_repo, opts.template) {
+        // given template is a remote git repository & needs to be cloned
+        // This will be cloned to .cargo/templates/<repo_name> where <repo_name>
```

Comment:
This doesn't seem to be true anymore, I think that's a left over from previous iterations.